cmd/deploy.go

```diff
@@ -177,31 +177,43 @@ func (dc *deployCmd) validate(cmd *cobra.Command, args []string) {
 			log.Fatal("apimodel invalid: ServicePrincipalProfile is missing either the clientid or secret.")
 		}
 
-		log.Warnln("apimodel: ServicePrincipalProfile was empty, creating application...")
+		if dc.containerService.Properties.OrchestratorProfile != nil &&
+			dc.containerService.Properties.OrchestratorProfile.KubernetesConfig != nil &&
+			dc.containerService.Properties.OrchestratorProfile.KubernetesConfig.UseManagedIdentity {
+			log.Warnln("apimodel: ServicePrincipalProfile was empty, but user is using msi. Skipping ServicePrincipal creation.")
+
+			dc.containerService.Properties.ServicePrincipalProfile = &api.ServicePrincipalProfile{
+				ClientID: "msi",
```

Comment:
Added a comment in the other PR. I don't think ServicePrincipalProfile fields should accept "msi" as a valid value. It confusing from end user and documentation standpoint.